### ISSUE TITLE
Form multi-node clusters in packaging tests

### ISSUE THREAD

**Issue by DaveCTurner** [elastic/elasticsearch#34809]
Today we only start a single node in our packaging tests. However, an important part of our out-of-the-box experience is that it's possible to start multiple nodes on a single host and have them form a cluster with very little extra configuration.

We currently test this in `SettingsBasedHostProviderIT#testClusterFormsByScanningPorts` but this is unreliable (see #34781) because it's not always possible to grab two free ports that are close enough together for the port scanning to work.

It seems that one easy way to do this would be to extract multiple installations using this:

https://github.com/elastic/elasticsearch/blob/0f1a5ecf6118e939a472331537f96d55f6fddda3/qa/vagrant/src/main/java/org/elasticsearch/packaging/util/Archives.java#L66

Then start them using this:

https://github.com/elastic/elasticsearch/blob/0f1a5ecf6118e939a472331537f96d55f6fddda3/qa/vagrant/src/main/java/org/elasticsearch/packaging/util/Archives.java#L257

It may also be desirable to start multiple nodes from the same installation, either by setting `max_local_storage_nodes` or by giving them different data paths, since these are all valid ways to start multi-node clusters on a single host.

**Comment by rjernst** [comment]
Having this be part of packaging tests so it is tested for rpm and deb as well would be good. Even better would be finding a way to have the "instructions" on what needs to be changed be used in the test. Maybe @nik9000 has ideas on that; something similar to doc snippets?

**Comment by mark-vieira** [comment]
> However, an important part of our out-of-the-box experience is that it's possible to start multiple nodes on a single host and have them form a cluster with very little extra configuration.

@DaveCTurner is this still the case given we now [explicitly suggest](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup.html#dedicated-host) that folks **do not** run multiple instances on a single host.

**Comment by DaveCTurner** [comment]
We say not to do that _in production_ but I believe it's still useful for new users to be able to run a few nodes locally and have everything just work. I don't hold that opinion very strongly tho, and I wouldn't argue against dropping it if we made a positive decision to do so. That'd mean we could also drop the default scan-for-other-nodes-on-nearby-ports behaviour.

**Comment by mark-vieira** [comment]
Yeah, I think it's worth being more deliberate here on what use cases we want to support, and how best to support them as well as the backing test coverage for those cases. Sounds like that's not well defined. This kind of thing obviously is going to depend _a lot_ on platform and packaging specific stuff. I think right now the [most documented way](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html#docker-file) to do this is via docker compose.

**Comment by DaveCTurner** [comment]
On deeper inspection it turns out that this use-case is already totally broken. For starters each node will default to consuming 50% of available RAM so (at least on my 64GB-RAM-no-swap Linux machine) if you start two bare nodes with completely default configs then one of them will explode. And then in 8.0 security is enabled by default, so even if you have enough memory the nodes won't trust each other and will form independent clusters. Given that you now always have to go through some ceremony to start multiple bare nodes, this issue is now moot so I'm closing it.
